server/grpc_service.go

```diff
@@ -1633,8 +1617,52 @@ func (s *GrpcServer) SplitRegions(ctx context.Context, request *pdpb.SplitRegion
 }
 
 // SplitAndScatterRegions split regions by the given split keys, and scatter regions
-func (s *GrpcServer) SplitAndScatterRegions(_ context.Context, _ *pdpb.SplitAndScatterRegionsRequest) (*pdpb.SplitAndScatterRegionsResponse, error) {
-	panic("unimplemented")
+func (s *GrpcServer) SplitAndScatterRegions(ctx context.Context, request *pdpb.SplitAndScatterRegionsRequest) (*pdpb.SplitAndScatterRegionsResponse, error) {
+	fn := func(ctx context.Context, client *grpc.ClientConn) (interface{}, error) {
+		return pdpb.NewPDClient(client).SplitAndScatterRegions(ctx, request)
+	}
+	if rsp, err := s.unaryMiddleware(ctx, request.GetHeader(), fn); err != nil {
+		return nil, err
+	} else if rsp != nil {
+		return rsp.(*pdpb.SplitAndScatterRegionsResponse), err
+	}
+	rc := s.GetRaftCluster()
+	splitFinishedPercentage, newRegionIDs := rc.GetRegionSplitter().SplitRegions(ctx, request.GetSplitKeys(), int(request.GetRetryLimit()))
+	scatterFinishedPercentage, err := scatterRegions(rc, newRegionIDs, request.GetGroup(), int(request.GetRetryLimit()))
+	if err != nil {
+		return nil, err
+	}
+	return &pdpb.SplitAndScatterRegionsResponse{
+		Header:                    s.header(),
+		RegionsId:                 newRegionIDs,
+		SplitFinishedPercentage:   uint64(splitFinishedPercentage),
+		ScatterFinishedPercentage: uint64(scatterFinishedPercentage),
+	}, nil
+}
+
+// scatterRegions add operators to scatter regions and return the processed percentage and error
+func scatterRegions(cluster *cluster.RaftCluster, regionsID []uint64, group string, retryLimit int) (int, error) {
+	ops, failures, err := cluster.GetRegionScatter().ScatterRegionsByID(regionsID, group, retryLimit)
+	if err != nil {
+		return 0, err
+	}
+	for _, op := range ops {
+		if ok := cluster.GetOperatorController().AddOperator(op); !ok {
+			failures[op.RegionID()] = fmt.Errorf("region %v failed to add operator", op.RegionID())
+		}
+	}
+	percentage := 100
+	if len(failures) > 0 {
```

Comment:
How about using a variable for `len(failures)`?

Reply:
pardon? I see many palce used like this for example handle
![image](https://user-images.githubusercontent.com/38561029/162686311-17d1ad67-094b-4d94-a3ab-bb8cd4bc60a0.png)

Reply:
maybe we can abstract a function to calculate the percentage?

Reply:
there is no raft cluster in scatterRegion. If we want abstract a function to calculate.,we have to change cluster to raftcluster in ReionScatterer
![image](https://user-images.githubusercontent.com/38561029/162697639-e37674bf-24d2-4e62-aeb8-8334d7b7511d.png)
but I found

Reply:
Got it. We can leave it for now.